firebase-perf/src/test/java/com/google/firebase/perf/transport/RateLimiterTest.java

```diff
@@ -398,6 +398,67 @@ public void testDeviceSampling_tracesDisabledButNetworkEnabled() {
     assertThat(limiter.getIsDeviceAllowedToSendTraces()).isFalse();
   }
 
+  @Test
+  public void testDeviceSampling_tracesEnabledButFragmentDisabled() {
+    makeConfigResolverReturnDefaultValues();
+    when(mockConfigResolver.getTraceSamplingRate()).thenReturn(0.5f);
+    when(mockConfigResolver.getFragmentSamplingRate()).thenReturn(0.02f);
+
+    RateLimiter limiter =
+        new RateLimiter(TWO_TOKENS_PER_MINUTE, 2, mClock, 0.49f, mockConfigResolver);
+
+    assertThat(limiter.getIsDeviceAllowedToSendFragmentScreenTraces()).isFalse();
+    assertThat(limiter.getIsDeviceAllowedToSendTraces()).isTrue();
+
+    // Fragment trace should be dropped
+    PerfMetric fragmentTrace =
+        PerfMetric.newBuilder()
+            .setTraceMetric(
+                TraceMetric.newBuilder()
+                    .setName("_st_TestFragment")
+                    .putCustomAttributes(Constants.ACTIVITY_ATTRIBUTE_KEY, "TestActivity")
+                    .addAllPerfSessions(Arrays.asList(createNonVerbosePerfSessions())))
+            .build();
+    assertThat(limiter.isFragmentScreenTrace(fragmentTrace)).isTrue();
+    assertThat(limiter.isEventSampled(fragmentTrace)).isFalse();
+
+    // Non-fragment trace should be sampled
+    PerfMetric activityTrace =
+        PerfMetric.newBuilder()
+            .setTraceMetric(
+                TraceMetric.newBuilder()
+                    .setName("_st_TestActivity")
+                    .addAllPerfSessions(Arrays.asList(createNonVerbosePerfSessions())))
+            .build();
+    assertThat(limiter.isFragmentScreenTrace(activityTrace)).isFalse();
+    assertThat(limiter.isEventSampled(activityTrace)).isTrue();
+  }
+
+  @Test
+  public void testDeviceSampling_tracesDisabledButFragmentEnabled() {
```

Comment:
nit: add expected behavior to the test name

Reply:
Done

Reply:
Is the test name updated?

Reply:
yes, all newly added tests should be using the naming convention